### PR TITLE
Add nxos_bgp_address_family Resource Module

### PR DESCRIPTION
**Pull request by NilashishC**
##### SUMMARY
- Targets https://github.com/ansible-network/resource_module_models/pull/112
- Add nxos_bgp_address_family RM

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
nxos_bgp_address_family.py

### CODE REVIEW

**Comment by NilashishC** [comment]
recheck

**Comment by NilashishC** [comment]
recheck

**Comment by NilashishC** [comment]
Module | coverage
-- | -- |
plugins/module_utils/network/nxos/argspec/bgp_address_family/bgp_address_family.py | 100%
plugins/module_utils/network/nxos/config/bgp_address_family/bgp_address_family.py | 99%
plugins/module_utils/network/nxos/facts/bgp_address_family/bgp_address_family.py | 98%
plugins/module_utils/network/nxos/rm_templates/bgp_address_family.py | 100%
plugins/modules/nxos_bgp_address_family.py | 88%
tests/unit/modules/network/nxos/test_nxos_bgp_address_family.py | 98%
Total | **98%**

File: plugins/module_utils/network/nxos/facts/bgp_address_family/bgp_address_family.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            nbr = []
+            if "address_family" in objs:
+                # remove neighbor AF entries
+                for k, v in iteritems(objs["address_family"]):
+                    if not k.startswith("nbr_"):
+                        nbr.append(k)
+                for x in nbr:
+                    del objs["address_family"][x]
+                objs["address_family"] = list(objs["address_family"].values())
+                # TO-DO sort all list of dictionaries
```
Comment (rohitthakur2590) [inline plugins/module_utils/network/nxos/facts/bgp_address_family/bgp_address_family.py:78]: looks like some expected modification.

File: plugins/module_utils/network/nxos/rm_templates/bgp_address_family.py [hunk 2]
```diff
@@ -0,0 +1,782 @@
...
+                            "trigger_delay": {
+                                "critical_delay": "{{ critical_delay }}",
+                                "non_critical_delay": "{{ non_critical_delay }}",
+                            }
+                        }
+                    }
+                }
+            },
+        },
+        # to do
```
Comment (rohitthakur2590) [inline plugins/module_utils/network/nxos/rm_templates/bgp_address_family.py:630]: Do we need something here ?

Comment (NilashishC) [reply plugins/module_utils/network/nxos/rm_templates/bgp_address_family.py:630]: Oh no, this one is done. Forgot to remove it. :)

File: plugins/module_utils/network/nxos/facts/bgp_address_family/bgp_address_family.py (hunk 1 above)
Comment (NilashishC) [reply plugins/module_utils/network/nxos/facts/bgp_address_family/bgp_address_family.py:78]: Yeah, I kept a reminder to sort every list of dictionary. Will push a commit.

File: plugins/module_utils/network/nxos/rm_templates/bgp_address_family.py [hunk 3]
```diff
@@ -0,0 +1,782 @@
...
+
+
+def _tmplt_aggregate_address(aggaddr):
+    cmd = "aggregate-address {prefix}"
+
+    if aggaddr.get("advertise_map"):
+        cmd += " advertise-map {advertise_map}"
+    if aggaddr.get("as_set"):
+        cmd += " as-set"
+    if aggaddr.get("attribute_map"):
```
Comment (ashwini-mhatre) [inline plugins/module_utils/network/nxos/rm_templates/bgp_address_family.py:30]: this condition is duplicated

**Review by rohitthakur2590 — APPROVED** [review]

**Pull request merged**